### DISCUSSION TITLE
[Feature] Allow adding of local assets to album triggering upload

### DISCUSSION THREAD

**Discussion by hrdl-github** [immich-app/immich#8595] [Feature Request]
### The feature

At the moment adding resources that are not backed up is a two-stage process: first they have to be uploaded, then they have to be selected again and added to an album. This is inconvenient and error-prone. Being able to add local assets to a (potentially new) album or sharing them using immich would be a nice feature for those who do not wish to back up all relevant assets automatically.

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
Duplicate of #1678

**Reply by hrdl-github** [reply]
I would argue that they are related, but #1678 focuses on local albums, while this issue is about remote albums and should be easier to implement.

**Reply by hrdl-github** [reply]
I think this request is still relevant despite #1678 being closed and I haven't found it being discussed elsewhere. When trying to add assets that are available only locally, the error case https://github.com/immich-app/immich/blob/930df46f74c0c9f543c919fa4a641d9c29f94db1/mobile/lib/widgets/asset_grid/multiselect_grid.dart#L265-L269 occurs.

**Reply by hrdl-github** [reply]
This is especially relevant in two cases:
1. Albums or users who do not want to use automatic synchronisation, e.g. because they are using somebody else's instance.
2. When adding assets to an album that haven't been synchronised yet, either due to recency or network conditions.

**Reply by hrdl-github** [reply]
Would you mind reopening this request, @bo0tzz as it is not a duplicate? I just thought that reopening this was better than creating a duplicate feature request.

**Reply by bo0tzz** [reply]
You're right. By now there's also #12190, which I think is a little clearer so I'll leave this closed in favour of that one.
